### DISCUSSION TITLE
[ALG101] 想請問LIOJ 1016的寫法改善

### DISCUSSION THREAD

**Discussion by wellan-H** [Lidemy/forum#132] [前端相關問題]
如題，與示範的不同，我是以第一個元素作為比較對象，以for跑完之後再輸出，目前知道可能會出現的額外錯誤有比如不同於題目的選項(如'C')，想請問是否有改善的地方?

function solve(lines){
    let num = lines[0]
    let group = ''
    for(let i = 0;i<=num;i++){
        group+=lines[i]
    }
    let tar =group[1]
    let count = 0
    let tarIndex =''
    let tar2Index = ''
    for(let i =1;i<group.length;i++){
        if(group[i]==tar){
            tarIndex+=(i+'\n')
            count++
        }else{
            tar2Index+=(i+'\n')
        }
    }
    let count2=num-count
    if(count<count2){
        console.log(tarIndex)
    }else if(count>count2){
        console.log(tar2Index)
    }else console.log("PEACE")

}

**Comment by allenliaooooooo** [comment]
先幫你排個版，方便大家閱讀
```js
function solve(lines){
  let num = lines[0]
  let group = ''
  for(let i = 0; i <= num; i++){
    group += lines[i]
  }
  let tar = group[1]
  let count = 0
  let tarIndex = ''
  let tar2Index = ''
  for(let i = 1; i < group.length; i++){
    if(group[i] == tar){
      tarIndex += (i + '\n')
      count++
    }else{
      tar2Index += (i + '\n')
    }
  }
  let count2 = num - count
  if(count < count2){
    console.log(tarIndex)
  }else if(count > count2){
    console.log(tar2Index)
  }else {
    console.log("PEACE")
  }
}
```

**Reply by allenliaooooooo** [reply]
> `let tar = group[1]`

這邊好像是用第二個元素做比較對象耶

**Reply by wellan-H** [reply]
不好意思沒說明清楚，我是直接拿整個題目下去跑，第0個是資料的數量，由索引值1開始比較的

**Reply by allenliaooooooo** [reply]
喔喔，是我自己沒看清楚 哈哈

**Comment by yichennnn36** [comment]
如果每個人都出一樣的也應該要跑出 PEACE～
